Hey @naveenkaushik2504 ,

I'll take a look at the code itself ASAP, but in the meanwhile please note that it fails the linting job on Travis:
https://travis-ci.org/github/pdpipe/pdpipe/builds/722429711

Here's the output:
```
./pdpipe/sklearn_stages.py:212:80: E501 line too long (87 > 79 characters)
./pdpipe/sklearn_stages.py:238:80: E501 line too long (87 > 79 characters)
./pdpipe/col_generation.py:271:80: E501 line too long (97 > 79 characters)
./pdpipe/col_generation.py:695:80: E501 line too long (89 > 79 characters)
./pdpipe/col_generation.py:888:80: E501 line too long (88 > 79 characters)
./pdpipe/col_generation.py:903:80: E501 line too long (99 > 79 characters)
```

--- a/pdpipe/col_generation.py
+++ b/pdpipe/col_generation.py
@@ -268,7 +268,8 @@ class OneHotEncode(ColumnsBasedPipelineStage):
             except KeyError:  # pragma: no cover
                 raise PipelineApplicationError((
                     "Missing encoder for column {} when applying a fitted "
-                    "OneHotEncode pipeline stage by class {} !").format(colname, self.__class__))
+                    "OneHotEncode pipeline stage by class {} !")
+                    .format(colname, self.__class__))
             res_cols = col.apply(encoder)
             for res_col in res_cols:
                 assign_map[res_col] = res_cols[res_col]
@@ -692,7 +693,8 @@ class ColByFrameFunc(PdPipelineStage):
             new_col = self._func(df)
         except Exception:
             raise PipelineApplicationError(
-                "Exception raised applying function {} to dataframe by class {}.".format(
+                "Exception raised applying function {} to "
+                "dataframe by class {}.".format(
                     self._func_desc, self.__class__
                 )
             )
@@ -885,7 +887,8 @@ class Log(ColumnsBasedPipelineStage):
             except KeyError:  # pragma: no cover
                 raise PipelineApplicationError((
                     "Missig column {} when applying a fitted "
-                    "Log pipeline stage by class {} !").format(colname, self.__class__))
+                    "Log pipeline stage by class {} !").format(
+                        colname, self.__class__))
             loc = df.columns.get_loc(colname) + 1
             new_name = colname + "_log"
             if self._drop:
@@ -900,7 +903,8 @@ class Log(ColumnsBasedPipelineStage):
                 else:  # pragma: no cover
                     raise PipelineApplicationError((
                         "Missig fitted parameter for column {} when applying a"
-                        " fitted Log pipeline stage by class {}!").format(colname, self.__class__))
+                        " fitted Log pipeline stage by class {}!").format(
+                            colname, self.__class__))
             # must check not None as neg numbers eval to False
             if self._const_shift is not None:
                 new_col = new_col + self._const_shift

--- a/pdpipe/sklearn_stages.py
+++ b/pdpipe/sklearn_stages.py
@@ -209,7 +209,8 @@ class Scale(ColumnsBasedPipelineStage):
             )
         except Exception:
             raise PipelineApplicationError(
-                "Exception raised when Scale applied to columns {} by class {}".format(
+                "Exception raised when Scale applied to columns"
+                " {} by class {}".format(
                     self._columns_to_scale, self.__class__
                 )
             )
@@ -235,7 +236,8 @@ class Scale(ColumnsBasedPipelineStage):
             )
         except Exception:
             raise PipelineApplicationError(
-                "Exception raised when Scale applied to columns {} by class {}".format(
+                "Exception raised when Scale applied to columns"
+                " {} by class {}".format(
                     self._columns_to_scale, self.__class__
                 )
             )

--- a/tests/core/test_pipeline.py
+++ b/tests/core/test_pipeline.py
@@ -7,7 +7,8 @@ from pdpipe.core import (
     PdPipelineStage,
     PdPipeline
 )
-from pdpipe import make_pdpipeline
+from pdpipe import make_pdpipeline, ColByFrameFunc, ColDrop
+from pdpipe.exceptions import PipelineApplicationError
 
 
 def _test_df():
@@ -185,3 +186,25 @@ def test_pipeline_slice():
     assert 'num1' not in res_df.columns
     assert 'num2' not in res_df.columns
     assert 'char' in res_df.columns
+
+@pytest.mark.parametrize("time", [True, False])
+def test_pipeline_error(time):
+    """Test exceptions at pipeline level"""
+
+    # test fit
+    df = _test_df()
+    func = lambda df: df['num1'] == df['num3']
+    pipeline = PdPipeline([ColByFrameFunc("Equality", func), ColDrop("B")])
+    with pytest.raises(PipelineApplicationError):
+        pipeline.fit(df, verbose=True, time=time)
+
+    # test transform
+    df = _test_df()
+    with pytest.raises(PipelineApplicationError):
+        pipeline.transform(df, verbose=True, time=time)
+
+    # test fit_transform
+    df = _test_df()
+    with pytest.raises(PipelineApplicationError):
+        pipeline.fit_transform(df, verbose=True, time=time)
+    

